pycbc/window.py

```diff
@@ -0,0 +1,665 @@
+        data will be whitened; if 2, data will be over-whitened; if 0 or False,
+        no whitening will be done. Default is False. If 1 or 2, psds must not
+        be None.
+    psds : (dict of) FrequencySeries, optional
+        Required if `window_whitened` is 1 or 2. Either a FrequencySeries or a
+        dictionary of FrequencySeries to use for whitening the data. If a
+        dictionary, must be `detector name -> psd`. Default is None.
+
+    Raises
+    ------
+    WindowBoundsError
+        Raised when `apply_window` is called and the requested left (right)
+        time of the window occurs after (before) the end (start) of the data,
+        such that the entire data segment would be zeroed.
+
+    Examples
+    --------
+    Create a window that uses a 5 second Hann window on the left and a 4
+    second Kaiser window with `beta = 8` on the right:
+
+    >>> from pycbc import window
+    >>> windowts = window.TimeDomainWindow(left_taper='hann', right_taper=('kaiser', 8.), left_taper_duration=5., right_taper_duration=4.)
+
+    Create a 16s time series of ones, and apply the above window to it,
+    starting at 2s and ending at 15s:
+
+    >>> from pycbc import types
+    >>> ts = types.TimeSeries(numpy.ones(16), delta_t=1.)
+    >>> windowts(ts, left_time=2., right_time=15.).data
+    ArrayWithAligned([ 0.,  0.,  0.,  0.0954915,  0.3454915, 0.6545085,
+                       0.9045085,  1.,  1.,  1., 1.,  1.,  0.78875245,
+                       0.36897272,  0.08273982,  0. ])
+
+    Over-whiten and window a CBC waveform using a lal taper on the left and
+    a 10ms kaiser taper ending 20ms before coalescence on the right:
+
+    >>> from pycbc import psd, waveform
+    >>> hp, _ = waveform.get_td_waveform(approximant='SEOBNRv4', mass1=40., mass2=30., delta_t=1./2048, f_lower=15.)
+    >>> aligopsd = psd.aLIGOZeroDetHighPower(4*2048/2+1, 1./4, 10.)
+    >>> windowts = window.TimeDomainWindow(left_taper='lal', right_taper=('kaiser', 8), right_taper_duration=0.01, window_whitened=2, psds=aligopsd)
+    >>> hp = windowts(hp, right_time=len(hp)*hp.delta_t-0.02)
+
+    Apply the same taper to a frequency-domain waveform. Note that we have to
+    supply a `break_time` indicating the start of the data series to account
+    for the wrap around of the waveform:
+
+    >>> hp, _ = waveform.get_fd_waveform(approximant='IMRPhenomD', mass1=40., mass2=30., delta_f=1./4, f_lower=15.)
+    >>> break_time = 1.
+    >>> hp = windowts(hp, right_time=1./hp.delta_f-break_time-0.02, break_time=break_time)
+
+    See Also
+    --------
+    scipy.signal.get_window : function for generating windows
+    laltaper_timeseries : function that only applies the lal taper
+    """
+    def __init__(self, left_taper=None, right_taper=None,
+                 left_taper_duration=None, right_taper_duration=None,
+                 window_whitened=False, psds=None):
+        if int(window_whitened) not in [UNWHITENED, WHITENED, OVERWHITENED]:
```

Comment:
Personally I would use a different approach for selecting whitening: an argument `whiten` which can be `None` (no whitening) or the strings `'whiten'` or `'overwhiten'`. So there is no need to define and import constants, and the code is more straightforward.